lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -3282,8 +3282,9 @@ def test_permuations(self):
                 assert show_courseware_mfe_link(global_staff_user, False, new_course_key)
                 assert show_courseware_mfe_link(regular_user, True, new_course_key)
 
-                # Regular users don't see the link.
-                assert not show_courseware_mfe_link(regular_user, False, new_course_key)
+                # (Regular users would see the link, but they can't see the Legacy
+                #  experience, so it doesn't matter.)
+
```

Comment:
With my change, if a regular user managed to view the Legacy experience (in an MFE enabled course), they would see "View in New Experience".

This is an impossible situation, and handling it would make the implementation a bit more complex, so I just dropped these test assertions.